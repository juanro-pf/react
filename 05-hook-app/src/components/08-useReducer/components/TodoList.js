import React from 'react';
import { TodoListItem } from './TodoListItem';
import PropTypes from 'prop-types';

export const TodoList = ({todos, handleToggle, handleDelete}) => {
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map( (todo, i) => 
                    <TodoListItem
                        key= {todo.id}
                        todo= {todo}
                        i= {i}
                        handleToggle= {handleToggle}
                        handleDelete= {handleDelete}
                    />
                )
            }
        </ul>
    );
};

TodoList.propTypes= {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
};