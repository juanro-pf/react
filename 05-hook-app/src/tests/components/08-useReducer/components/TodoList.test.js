import React from 'react';
import { demoTodos } from '../../../fixtures/demoTodos';
import { shallow } from 'enzyme';
import { TodoList } from '../../../../components/08-useReducer/components/TodoList';

import '@testing-library/jest-dom';

describe('Testing TodoList', () => {
    
const handleToggle= jest.fn();
const handleDelete= jest.fn();

    const wrapper= shallow(
        <TodoList
        todos= {demoTodos}
        handleToggle= { handleToggle }
        handleDelete= { handleDelete }
        />
    );
    
    test('Snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('2 items', () => {

        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual( expect.any(Function) );

    });
    

});
