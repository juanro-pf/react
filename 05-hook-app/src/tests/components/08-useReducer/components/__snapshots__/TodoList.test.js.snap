// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoList Snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
